addons/l10n_in/models/account_invoice.py

```diff
@@ -122,3 +123,16 @@ def unlink(self):
         if logger_msg:
             _logger.info(logger_msg)
         return res
+
+    def _generate_qr_code(self, silent_errors=False):
+        self.ensure_one()
+        if self.company_id.country_code == 'IN':
+            payment_url = 'upi://pay?pa=%s&pn=%s&am=%s&tr=%s&tn=%s' % (
```

Comment:
Unrelated with this, but it is still weird there is no url_encode/quote needed for this, but I think you said it was ok this way.